.github/workflows/doc_build.yml

```diff
@@ -0,0 +1,52 @@
+name: Docs
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Setup Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: "3.10"
+          architecture: x64
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      - name: Install Dependencies
+        run: |
+          set -eux
+
+          pip install -r docs/requirements.txt
```

Comment:
Need to install requirements.txt from the root of the repo. 
and then build monarch from source: `pip install --no-build-isolation .`